agent/accumulator.go

```diff
@@ -111,7 +112,7 @@ func (ac *accumulator) AddError(err error) {
 		return
 	}
 	NErrors.Incr(1)
-	log.Printf("E! [%s]: Error in plugin: %v", ac.maker.Name(), err)
+	log.Printf("E! [%s]: Error in plugin: %v", ac.maker.LogName(), err)
```

Comment:
Perhaps I'm missing something, but I'm not seeing how this can be pulled out of metricmaker as several plugins call `acc.AddError` directly, if we remove `LogName` from metricmaker, those logs won't use the plugin alias.

Reply:
Yeah, how about instead of RunningInput having a `LogName()` function it has a `Logger() telegraf.Logger` function.  We then pass in a Logger to the Accumulator in agent/agent.go.

Reply:
Still need to do this, also need to try what we talked about with the reflection method for injecting the Logger.

Reply:
Oh wait I see the reflection stuff, so we just need to pull PluginConfig out.

Reply:
Oops, looks like I did forget to pull out PluginConfig from the plugin readme's